### ISSUE TITLE
Infer.Fields ignores field boost

### ISSUE THREAD

**Issue by robertlyson** [elastic/elasticsearch-net#4215]
Hi 👋

**NEST/Elasticsearch.Net version**: 6.6.0 and 7.4.0

**Elasticsearch version**: 6.6.0 and 7.4.0

**Description of the problem including expected versus actual behavior**:
Issue reported in this [so ](https://stackoverflow.com/questions/58801023/elastic-search-nest-client-not-applying-boost) question.

Looks like when trying to create fields with the following syntax 
```
Fields = Infer.Fields<RecipeSearchModel>(
    f1 => Infer.Field<RecipeSearchModel>(f => f.Title, 5),
    f2 => f2.Description,
    f3 => Infer.Field<RecipeSearchModel>(f => f.Ingredients, 3),
    f4 => f4.Method,
    f5 => Infer.Field<RecipeSearchModel>(f => f.Image.Alt, 4))
```
NEST, somewhere down the path is ignoring boost settings and producing this request to elasticsearch
```
{
  "query": {
    "multi_match": {
      "fields": [
        "title",
        "description",
        "ingredients",
        "method",
        "image.alt"
      ],
      "fuzziness": "AUTO",
      "query": "query",
      "type": "best_fields"
    }
  }
}
```
When changing syntax to 
```
Fields = Infer.Fields<RecipeSearchModel>()
    .And(Infer.Field<RecipeSearchModel>(f => f.Title, 5))
    .And<RecipeSearchModel>(f => f.Description)
    .And(Infer.Field<RecipeSearchModel>(f => f.Ingredients, 3))
    .And<RecipeSearchModel>(f => f.Method)
    .And(Infer.Field<RecipeSearchModel>(f => f.Image.Alt, 4))
```
all seems to be generated fine
```
{
  "query": {
    "multi_match": {
      "fields": [
        "title^5",
        "description",
        "ingredients^3",
        "method",
        "image.alt^4"
      ],
      "fuzziness": "AUTO",
      "query": "query",
      "type": "best_fields"
    }
  }
}
```

**Steps to reproduce**:
Here is the sample app (NEST 6.6.0) demonstrating the issue
```
using System;
using System.Text;
using System.Threading.Tasks;
using Elasticsearch.Net;
using Nest;

namespace NestTest
{
    class Program
    {
        static async Task Main(string[] args)
        {
            var connectionPool = new SniffingConnectionPool(new[] {new Uri("http://localhost:9200")});
            var settings = new ConnectionSettings(connectionPool)
                .DefaultIndex("index_name")
                .EnableDebugMode()
                .DisableDirectStreaming()
                .PrettyJson();
            var client = new ElasticClient(settings);
            
            var queryContainer = new MultiMatchQuery
            {
                Fuzziness = Fuzziness.Auto,
                Query = "query",
                Type = TextQueryType.BestFields,
                Fields = Infer.Fields<RecipeSearchModel>(
                    f1 => Infer.Field<RecipeSearchModel>(f => f.Title, 5),
                    f2 => f2.Description,
                    f3 => Infer.Field<RecipeSearchModel>(f => f.Ingredients, 3),
                    f4 => f4.Method,
                    f5 => Infer.Field<RecipeSearchModel>(f => f.Image.Alt, 4))
            };

            var searchResponse = await client.SearchAsync<RecipeSearchModel>(s => s.Query(q => queryContainer));

            Console.WriteLine("Incorrect: ");
            Console.WriteLine(Encoding.UTF8.GetString(searchResponse.ApiCall.RequestBodyInBytes));
            
            Console.WriteLine();
            
            queryContainer = new MultiMatchQuery
            {
                Fuzziness = Fuzziness.Auto,
                Query = "query",
                Type = TextQueryType.BestFields,
                Fields = Infer.Fields<RecipeSearchModel>()
                    .And(Infer.Field<RecipeSearchModel>(f => f.Title, 5))
                    .And<RecipeSearchModel>(f => f.Description)
                    .And(Infer.Field<RecipeSearchModel>(f => f.Ingredients, 3))
                    .And<RecipeSearchModel>(f => f.Method)
                    .And(Infer.Field<RecipeSearchModel>(f => f.Image.Alt, 4))
            };

            searchResponse = await client.SearchAsync<RecipeSearchModel>(s => s.Query(q => queryContainer));
                
            Console.WriteLine("Correct: ");
            Console.WriteLine(Encoding.UTF8.GetString(searchResponse.ApiCall.RequestBodyInBytes));
        }
    }

    public class RecipeSearchModel
    {
        public string Ingredients { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string Method { get; set; }
        public Image Image { get; set; }
    }

    public class Image
    {
        public string Alt { get; set; }
    }
}
```
Output:
```
Incorrect:
{
  "query": {
    "multi_match": {
      "fields": [
        "title",
        "description",
        "ingredients",
        "method",
        "image.alt"
      ],
      "fuzziness": "AUTO",
      "query": "query",
      "type": "best_fields"
    }
  }
}

Correct:
{
  "query": {
    "multi_match": {
      "fields": [
        "title^5",
        "description",
        "ingredients^3",
        "method",
        "image.alt^4"
      ],
      "fuzziness": "AUTO",
      "query": "query",
      "type": "best_fields"
    }
  }
}

```

Tested with latest version of NEST (7.4.0) and the same issue is happening.

**Provide `ConnectionSettings` (if relevant)**:

**Provide `DebugInformation` (if relevant)**:

**Describe the feature**:

**Comment by Mpdreamz** [comment]
Hi! 👋 

Interestingly the following syntax won't compile for me:

```csharp
Fields<Project>(
	f1 => Field<Project>(f => f.Description, 2.2),
	f2 => f2.Name
),
```
> [CS0854] An expression tree may not contain a call or invocation that uses optional arguments

That fields overload taking  `Expression[]` is simply meant to dispatch on the first arg e.g `f1,f2,f3` 

```csharp
.And<Project, string>(p=>p.Description, 2.2)
```

or more involved perhaps

```cshap
Fields<Project>(
	f1 => f1.Description.Suffix("^2.2"),
	f2 => f2.Name
)
```

Are other approaches append a boost. 

Sadly you need to specify `TValue` otherwise it assumes the `.And` overload taking many `Expression[]`, I am keen to rename this overload to `.AndMany()` though in 8.0.

I think the best thing for us to do is to be more strict with our ExpressionResolver and mandate the argument is used.

**Comment by robertlyson** [comment]
> Interestingly the following syntax won't compile for me:
> 
> Fields<Project>(
> 	f1 => Field<Project>(f => f.Description, 2.2),
> 	f2 => f2.Name
> ),

At first glance, I couldn't understand how it doesn't compile for you if it does for me and Fishcake. Then I checked c# version for my test project and it was set to 7.3 Out of curiosity I changed it to 8.0 and it stopped compiling. I was like "did something changed in c# 8.0 :thinking:", so I switched back to 7.3 but it just stopped working :smiley: I don't know, maybe my Rider installation was messing with me yesterday, but I'm quite sure it was compiling. 

Nevermind then. If that wasn't meant to be working in the first place I think we can close the issue. Thanks for spending your time looking into it :+1:
